src/pipelinedefinition.cpp

```diff
@@ -517,8 +537,13 @@ class NodeValidator {
     }
 
     Status validate() {
+        auto result = checkForRestrictedBatchSize();
```

Comment:
Move to line L550 so you do not need to check in `checkForRestrictedBatchSize` if such model exists. You will also not need to call findModelByName, because we already call `fetchUnderlyingModelInstance` in L546